Lombiq.Analyzers.PowerShell/Rules/Measure-VariableNameFormat/Measure-VariableNameFormat.psm1

```diff
@@ -0,0 +1,344 @@
+    - PSUseCorrectVariableNameCasing: Variable names should start with a lowercase letter.
+    - PSUseParametersAsDeclared: Parameters should be used with exactly as they were declared.
+
+    When fixing warnings, work through the rules in the order they are listed above, because violating
+    PSAvoidUsingUnnecessaryBracesInVariableNames and/or PSUseCorrectParameterNameCasing, while referencing that
+    parameter with the correct casing or format, will also raise a PSUseParametersAsDeclared warning. The latter will
+    disappear after fixing the former(s).
+.EXAMPLE
+    Measure-VariableNameFormat -Ast $Ast
+.INPUTS
+    [System.Management.Automation.Language.ScriptBlockAst]
+.OUTPUTS
+    [Microsoft.Windows.Powershell.ScriptAnalyzer.Generic.DiagnosticRecord[]]
+#>
+
+using namespace System.Management.Automation.Language
+
+Import-Module (Join-Path (Split-Path -Path $MyInvocation.MyCommand.Path) '..\AstFunctions.ps1') -Force
+
+function Measure-VariableNameFormat
+{
+    [CmdletBinding()]
+    [OutputType([Microsoft.Windows.Powershell.ScriptAnalyzer.Generic.DiagnosticRecord[]])]
+    Param
+    (
+        [Parameter(Mandatory = $true)]
+        [ValidateNotNullOrEmpty()]
+        [System.Management.Automation.Language.ScriptBlockAst]
+        $Ast
+    )
+
+    Process
+    {
+        $analyzerViolations = New-Object System.Collections.Generic.List[Microsoft.Windows.Powershell.ScriptAnalyzer.Generic.DiagnosticRecord]
+
+        # This analyzer can and should analyze the whole file only, especially if it has embedded functions, because it
+        # needs to keep track of parameters declared in the root and the parent functions of a given function.
+        if ($Ast.Extent.StartLineNumber -ne 1 -or $Ast.Extent.StartColumnNumber -ne 1)
+        {
+            return $analyzerViolations
+        }
+
+        $correctionTypeName = 'Microsoft.Windows.PowerShell.ScriptAnalyzer.Generic.CorrectionExtent'
+        # See https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_automatic_variables.
+        $automaticVariableNames = (
+            '$$', '$?', '$^', '$_', '$args', '$ConsoleFileName', '$EnabledExperimentalFeatures', '$Error', '$Event',
+            '$EventArgs', '$EventSubscriber', '$ExecutionContext', '$false', '$foreach', '$HOME', '$Host', '$input',
+            '$IsCoreCLR', '$IsLinux', '$IsMacOS', '$IsWindows', '$LASTEXITCODE', '$Matches', '$MyInvocation',
+            '$NestedPromptLevel', '$null', '$PID', '$PROFILE', '$PSBoundParameters', '$PSCmdlet', '$PSCommandPath',
+            '$PSCulture', '$PSDebugContext', '$PSEdition', '$PSHOME', '$PSItem', '$PSScriptRoot', '$PSSenderInfo',
+            '$PSUICulture', '$PSVersionTable', '$PWD', '$Sender', '$ShellId', '$StackTrace', '$switch', '$this', '$true')
+
+        try
+        {
+            # The whole script block being analyzed is the root, which we call '€$', since it doesn't have a name.
```

Comment:
Is this `€$` coming from somewhere or just an arbitrary string?

Reply:
Arbitrary. I started with just `$` and then got inspired by https://cyberpunk.fandom.com/wiki/Eurodollar. :D

Reply:
Ah, I never realized that's where "eddie" comes from! Cool. :grin: 

Please make it clear in the comment that this is an arbitrary unique string. Otherwise the reader may think it's a magic string with special significance in pwsh. (there are such things in other shells so it's a reasonable assumption)

Reply:
Updated, although the € has no syntactical meaning in PS and it's also just a constant string.